packages/flutter/lib/src/widgets/sliver.dart

```diff
@@ -1907,6 +1911,470 @@ class KeepAlive extends ParentDataWidget<KeepAliveParentDataMixin> {
+  /// is canceled or after done and the sliver retracted away.
+  inactive,
+
+  /// While being overscrolled, but not far enough yet to trigger the refresh.
+  drag,
+
+  /// Dragged far enough that the onRefresh callback will run and the dragged
+  /// displacement is not yet at the final refresh resting state.
+  armed,
+
+  /// While the onRefresh task is running.
+  refresh,
+
+  /// While the indicator is animating away after refreshing.
+  done,
+}
+
+/// A base [StatefulWidget] class for sliver widget pull to refresh content control.
+///
+/// Concrete subclasses must override [buildRefreshIndicator].
+///
+/// See also:
+///
+/// * [SliverRefreshControl], A sliver refresh control, that reflects the standard Material pull-to-refresh behavior.
+/// * [CupertinoSliverRefreshControl], A sliver refresh control, that reflects the standard iOS pull-to-refresh behavior.
+///
+abstract class BaseSliverRefreshControl extends StatefulWidget {
+  /// An abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const BaseSliverRefreshControl({
+    super.key,
+    this.refreshTriggerPullDistance = _defaultRefreshTriggerPullDistance,
+    this.refreshIndicatorExtent = _defaultRefreshIndicatorExtent,
+    this.builder,
+    this.onRefresh,
+  }) : assert(refreshTriggerPullDistance != null),
+       assert(refreshTriggerPullDistance > 0.0),
+       assert(refreshIndicatorExtent != null),
+       assert(refreshIndicatorExtent >= 0.0),
+       assert(
+         refreshTriggerPullDistance >= refreshIndicatorExtent,
+         'The refresh indicator cannot take more space in its final state '
+         'than the amount initially created by overscrolling.',
+       );
+
+  /// The amount of overscroll the scrollable must be dragged to trigger a reload.
+  ///
+  /// Must not be null, must be larger than 0.0 and larger than
+  /// [refreshIndicatorExtent]. Defaults to 100px when not specified.
+  ///
+  /// When overscrolled past this distance, [onRefresh] will be called if not
+  /// null and the [builder] will build in the [RefreshIndicatorMode.armed] state.
+  final double refreshTriggerPullDistance;
+
+  /// The amount of space the refresh indicator sliver will keep holding while
+  /// [onRefresh]'s [Future] is still running.
+  ///
+  /// Must not be null and must be positive, but can be 0.0, in which case the
+  /// sliver will start retracting back to 0.0 as soon as the refresh is started.
+  /// Defaults to 60px when not specified.
+  ///
+  /// Must be smaller than [refreshTriggerPullDistance], since the sliver
+  /// shouldn't grow further after triggering the refresh.
+  final double refreshIndicatorExtent;
+
+  /// A builder that's called as this sliver's size changes, and as the state
+  /// changes.
+  ///
+  /// Can be set to null, in which case nothing will be drawn in the overscrolled
```

Comment:
This doesn't seem to be true. It falls back to `buildRefreshIndicator` in `_BaseSliverRefreshControlState.build`. Is there a reason you're making this change?